File: transaction_generator/location.py
from dataclasses import dataclass


@dataclass
class Location:
    latitude: float
    longitude: float

    def __str__(self):
        return f"{self.latitude}, {self.longitude}"

    def to_dict(self):
        return {"latitude": self.latitude, "longitude": self.longitude}

    def __iter__(self):
        yield "latitude", self.latitude
        yield "longitude", self.longitude

    @staticmethod
    def from_dict(location_dict):
        return Location(latitude=location_dict["latitude"], longitude=location_dict["longitude"])

File: anomaly_detector/src/main/java/psd/Location.java
package psd;

public class Location {
    private Double latitude;
    private Double longitude;

    public Location(Double latitude, Double longitude) {
        this.latitude = latitude;
        this.longitude = longitude;
    }

    @Override
    public String toString() {
        return "Location{" +
                "latitude=" + latitude +
                ", longitude=" + longitude +
                '}';
    }
}
